pages/benefit/[id].js

```diff
@@ -1,39 +1,60 @@
 import Head from "next/head";
 import { Banner } from "../../components/atoms/Banner";
+import { LanguageToggle } from "../../components/atoms/LanguageToggle";
 import { Benefit } from "../../components/organisms/Benefit.js";
+import { useTranslation } from "next-i18next";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 
 //TODO: replace lib/benefits with proper source of data
 import { getBenefitData, getBenefitIds } from "../../lib/benefits";
 
 //TODO: get benefitData from better source
-export async function getStaticProps({ params }) {
-  const benefitData = getBenefitData(params.id);
+export async function getStaticProps(context) {
+  const locale = context.locale || context.defaultLocale;
+  const benefitId = context.params.id;
   return {
     props: {
-      benefitData,
+      benefitId,
+      locale,
+      ...(await serverSideTranslations(locale, ["common"])),
     },
   };
 }
 
 //TODO: get paths data from better source
 export async function getStaticPaths() {
-  const paths = getBenefitIds();
+  //There is probably a better way to create all the paths needed
+  const benefitIds = getBenefitIds();
+  const pathsEn = benefitIds.map((x) => {
+    return { params: { id: x }, locale: "en" };
+  });
+  const pathsFr = benefitIds.map((x) => {
+    return { params: { id: x }, locale: "fr" };
+  });
+  const paths = pathsEn.concat(pathsFr);
+
   return {
     paths,
     fallback: false,
   };
 }
 
-export default function BenefitPage({ benefitData }) {
+export default function BenefitPage({ benefitId, locale }) {
+  const { t } = useTranslation("common");
+  const benefitData = getBenefitData(benefitId, t);
   return (
     <div className="mx-auto">
       <Head>
-        <title>{benefitData.name}</title>
+        <title>{benefitData.id}</title>
         <link rel="icon" href="/favicon.ico" />
       </Head>
-
+      <LanguageToggle
+        lang={locale}
+        languageToggleText={t("toggleLanguageText")}
+        queryValues={benefitId}
+      />
       <main className="mx-auto">
-        <Banner siteTitle={benefitData.name}></Banner>
+        <Banner siteTitle={benefitData.id} headline={benefitData.type}></Banner>
```

Comment:
Same problem here as the site title. I don't think ids make for good names.

Reply:
I changed it to title but I think we will still need to revisit when implementing Strapi